### PR TITLE
fix(BE): better handling PNG export failure

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

Fixing the displayerror method to avoid the blank page / crash of the browser
Enhancing messages displayed
Creating a PNG with the error message instead ot the RRD graph

**Fixes** # (https://github.com/centreon/centreon/pull/7822) requested changes

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/views/graphs/generateGraphs/generateMetricImage.php [hunk 1]
```diff
@@ -72,18 +72,15 @@
...
-    $row = $res->fetch();
-    if (!$row) {
+    try {
+        $res->bindValue(':hostId', $hostId, \PDO::PARAM_INT);
+        $res->bindValue(':svcId', $svcId, \PDO::PARAM_INT);
+        $res->execute();
+        $row = $res->fetch();
+        $index = $row['id'];
+    } catch (\PDOException $e) {
         CentreonGraph::displayError();
```
Comment (callapa) [inline www/include/views/graphs/generateGraphs/generateMetricImage.php]: Should the execution continue after?

Comment (sc979) [reply www/include/views/graphs/generateGraphs/generateMetricImage.php]: yes

File: www/class/centreonGraph.class.php [hunk 2]
```diff
@@ -1244,22 +1244,38 @@ public function addArgument($arg)
...
+        }
+
+        // generating the PNG with the error message
+        imagestring($image, 5, 0, 0, $str, $textcolor);
         header("Content-Type: image/gif");
 
-        imagegif($image);
-        exit;
+        // making the PNG to be saved instead of the graph
+        imagepng($image);
```
Comment (loiclau) [inline www/class/centreonGraph.class.php]: look the header

Comment (sc979) [reply www/class/centreonGraph.class.php]: done

**Pull request merged**